scripts/calculateRevenue/protocols/aave/blockchainData/subgraph.ts

```diff
@@ -0,0 +1,89 @@
+import { gql, GraphQLClient } from 'graphql-request'
+import { Address } from 'viem'
+import { SUBGRAPH_BASE_URL, THE_GRAPH_API_KEY } from '../config'
+import { BalanceSnapshot } from '../types'
+
+interface UserReserve {
+  reserve: {
+    aToken: {
+      id: string
+    }
+  }
+  aTokenBalanceHistory: {
+    timestamp: number
+    scaledATokenBalance: string
+    index: string
+  }[]
+}
+
+interface AaveUserReservesResponse {
+  userReserves: UserReserve[]
+}
+
+// Fetches the aToken scaled balance history from the Aave subgraph for a user address in a block range
+export async function getATokenScaledBalanceHistory({
+  subgraphId,
+  userAddress,
+  startTimestamp,
+  endTimestamp,
+}: {
+  subgraphId: string
+  userAddress: Address
+  startTimestamp: Date
+  endTimestamp: Date
+}): Promise<Map<Address, BalanceSnapshot[]>> {
+  const subgraphUrl = new URL(subgraphId, SUBGRAPH_BASE_URL).toString()
+
+  const client = new GraphQLClient(subgraphUrl, {
+    headers: {
+      Authorization: `Bearer ${THE_GRAPH_API_KEY}`,
+    },
+  })
+
+  const query = gql`
+    query getUserReservesHistory(
+      $userAddress: String!
+      $startTimestamp: Int!
+      $endTimestamp: Int!
+    ) {
+      userReserves(where: { user: $userAddress }) {
+        reserve {
+          aToken {
+            id
+          }
+        }
+        aTokenBalanceHistory(
+          where: {
+            timestamp_gte: $startTimestamp
+            timestamp_lte: $endTimestamp
```

Comment:
Depending on whether we actually follow the inclusive-start, exclusive-end convention for ranges, this should be `timestamp_lt`.